torch/csrc/autograd/engine.cpp

```diff
@@ -782,6 +782,20 @@ void Engine::evaluate_function(
     Node* func,
     InputBuffer& inputs,
     const std::shared_ptr<ReadyQueue>& cpu_ready_queue) {
+  // REVIEW QUESTION FOR ALBAN: Should this be up here, or down where it was originally?
```

Comment:
The one difference I see here is that the capture hooks will now have TLS fully set.
These are only used by the distributed engine right now, so @pritamdamania87 is that something you were expecting?

If it doesn't matter for distributed, I think we should move it up here indeed as I think users expect hooks to be called in the same context as the rest of the code.

Reply:
I think this should be fine for distributed.